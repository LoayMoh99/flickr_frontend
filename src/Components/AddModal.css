.list-label {
  padding: 0 20px 0 0;
  font-weight: 600;
}
.space-add {
  padding-left: 40px;
}
