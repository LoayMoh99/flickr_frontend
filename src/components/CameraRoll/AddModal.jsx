/* eslint-disable linebreak-style */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable linebreak-style */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { useState , useEffect } from 'react';
import axios from "axios"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import { faPlus } from '@fortawesome/free-solid-svg-icons';

import AddAlbumImages from './AddAlbumImages';
import './AddModal.css';
const endpoint = 'http://localhost:3001/'

function AddModal(props) {

  const [album,setAlbum] =useState([]);
  const  { imgAddIds }= props;

  //This function is called whenever this album page is opened
  //It gets the details of the selected album and set the const album with these details 
  useEffect(() => {
    const fetchData = async () => {
    const {data,status} = await axios.get( endpoint+'album',);
    console.log(status);
    console.log("got the album");
  
    if (status === 200){
        setAlbum(data);
    }
   
};

  fetchData();
},[album]);
console.log(album);

  // const albumCover='https://picsum.photos/id/237/200/300';
  //const search = <FontAwesomeIcon icon={faSearch} color="DarkGrey" />;
  const plus = <FontAwesomeIcon icon={faPlus} color="DarkGrey" />;



  return (
    <>

      <div className="modal__backdrop_edit">
        <div className="modalContainerAdd">
          <span className="closeAdd" onClick={props.onRequestAddClose}>&times;</span>
          <div className="row">
            <h3 className="modal__title_edit list-label">Add to: </h3>
            {/* <span className="space-add" /> */}
            <h3 className="modalAdd">Album </h3>
          </div>
          <hr id="hrSearch"/>
          {/* <div className="row">
            <div className="searchIcon">
             {search}
            </div>
          <input type="text" placeholder="Search albums" className="searchInput" tabIndex="0" />
          </div>
          <hr id="hrSearch"/> */}


          <div className="addBody">
            {album.map(album=>  
            (<AddAlbumImages 
            albumCover = {album.coverPhoto.photoUrl}
            albumTitle = {album.title}
            //photoCount = {album.photos.length}
            albumId ={album._id}
            albumPhotos = {album.photos}
            imgAddIds = {imgAddIds}
           />)
           ) 
            }
         </div>
         <hr id="hrBottom"/>
         <a href="/#" className="newAlbum" onClick={props.onRequestCreate}>
            {plus}
            {' '}
               Create new album
         </a>
         <button
              id="doneAdd"
              type="button"
              onClick={props.onRequestAddClose}
            >Done</button>
        </div>
      </div>
    </>
  );
}

export default AddModal;
