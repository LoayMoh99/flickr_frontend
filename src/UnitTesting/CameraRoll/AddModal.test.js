import React from 'react';
import ReactDOM from 'react-dom';
import AddModal from '../../components/CameraRoll/AddModal';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('AddModal renders without crashing', ()=>{
    const dummy =()=>{return 1;};
    const div = document.createElement('div');
    ReactDOM.render(<AddModal
        onRequestAddClose={dummy}
        onRequestCreate={dummy}
        imgAdd="[{_id:1},{_id:2}]"
        imgAddIds= "[1,2]"
        countAdd="2"  
        />, div);

});

it('AddModal renders correctly enzyme', () => {
    const dummy =()=>{return 1;};
    const wrapper = shallow(<AddModal
        onRequestAddClose={dummy}
        onRequestCreate={dummy}
        imgAdd="[{_id:1},{_id:2}]"
        imgAddIds= "[1,2]"
        countAdd="2"  
        />)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });
