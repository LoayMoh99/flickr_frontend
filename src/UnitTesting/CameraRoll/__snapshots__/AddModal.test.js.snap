// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddModal renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="modal__backdrop_edit"
  >
    <div
      className="modalContainerAdd"
    >
      <span
        className="closeAdd"
        onClick={[Function]}
      >
        Ã—
      </span>
      <div
        className="row"
      >
        <h3
          className="modal__title_edit list-label"
        >
          Add to: 
        </h3>
        <h3
          className="modalAdd"
        >
          Album 
        </h3>
      </div>
      <hr
        id="hrSearch"
      />
      <div
        className="addBody"
      />
      <hr
        id="hrBottom"
      />
      <a
        className="newAlbum"
        href="#"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          color="DarkGrey"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f067",
                "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
              ],
              "iconName": "plus",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        Create new album
      </a>
      <button
        id="doneAdd"
        onClick={[Function]}
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</Fragment>
`;
